tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/DAGIterator.java

```diff
@@ -124,6 +124,21 @@ private boolean advanceNextResponse() {
   }
 
   private boolean readNextRegionChunks() {
+    while (true) {
+      if (eof || regionTasks == null || taskIndex >= regionTasks.size()) {
+        return false;
+      }
+      if (doReadNextRegionChunks()) {
+        return true;
+      } // else {
+      // if doReadNextRegionChunks returns false
+      // readNextRegionChunks should not just return false
+      // readNextRegionChunks should read next region chunks
+      // }
+    }
+  }
+
+  private boolean doReadNextRegionChunks() {
     if (eof || regionTasks == null || taskIndex >= regionTasks.size()) {
```

Comment:
Do we need this if? It seems we did this predicate in `regionNextRegionChunks`

Reply:
yes, can delete

Reply:
done